homeassistant/components/shelly/utils.py

```diff
@@ -471,10 +471,9 @@ def is_rpc_wifi_stations_disabled(
     config: dict[str, Any], _status: dict[str, Any], key: str
 ) -> bool:
     """Return true if rpc all WiFi stations are disabled."""
-    if config[key]["sta"]["enable"] is True or config[key]["sta1"]["enable"] is True:
-        return False
-
-    return True
+    return (
+        config[key]["sta"]["enable"] is False and config[key]["sta1"]["enable"] is False
+    )
```

Comment:
Just as a sidenote, it should be considered per each case what is the better option. While here this condition is rarely used and can be written both in negative form and positive form it was written in a way that the first condition `if config[key]["sta"]["enable"] is True` will be true 99% of the time so that the other statements are not evaluated.

Specifically this case can be modified, but I am not familiar with all the other changes and they might have impact on performance.

Reply:
Yes, I rewrote it manually to avoid the double negation, but I forgot about the topic perfomance, you are right.